### PR TITLE
Particle terrain map

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Tests/Particles/Mapped/AMReX_TracerParticle_mod_K.H [hunk 1]
```diff
@@ -276,7 +276,8 @@ void linear_interpolate_to_particle_z (const P& p,
 
 #if (AMREX_SPACEDIM == 2)
 
-        int const j  = static_cast<int>(amrex::Math::floor(ly));
+        //int const j  = static_cast<int>(amrex::Math::floor(ly));
+        int const j  = p.idata(0);
```
Comment (RevathiJambunathan) [inline Tests/Particles/Mapped/AMReX_TracerParticle_mod_K.H]: this is what was wrong!!! 
Fixed now

File: Tests/Particles/Mapped/AMReX_TracerParticle_mod_K.H [hunk 2]
```diff
@@ -327,6 +345,7 @@ void linear_interpolate_to_particle_z (const P& p,
         amrex::Real const yint = ly - static_cast<Real>(j0);
         amrex::Real sx[] = { amrex::Real(1.) - xint, xint};
         amrex::Real sy[] = { amrex::Real(1.) - yint, yint};
+        k  = p.idata(0);
```
Comment (RevathiJambunathan) [inline Tests/Particles/Mapped/AMReX_TracerParticle_mod_K.H:326]: fixed for 3D terrain

File: Tests/Particles/Mapped/AMReX_TracerParticle_mod_K.H [hunk 3]
```diff
@@ -683,12 +702,12 @@ void linear_interpolate_to_particle_mapped (const P& p,
 
         amrex::Real f = b*x + c*y + d*x*y;
 
+        val[comp-start_comp] = f1 + f;
```
Comment (RevathiJambunathan) [inline Tests/Particles/Mapped/AMReX_TracerParticle_mod_K.H:538]: moved this here so it compiles with 3D

Comment (asalmgren) [reply Tests/Particles/Mapped/AMReX_TracerParticle_mod_K.H:538]: I thought what I committed last night compiled in 3D.  Are you seeing otherwise?

**Pull request closed without merging**